funmi5/ncc-media-team-backend
# #170071446 Build out CRUD endpoints for resources and songs

**What does this PR do?**
This PR builds out the CRUD endpoints for the resources

**Description of tasks**
Built out endpoints for users to add, get, delete, edit and update resources and songs

**How should this PR be tested/checked?**
Clone the project and run npm install
Run npm run start:dev
Make sure to update your .env file
To run the migration, run npm run migrate
To run the seeders, run npm run seed
To test: npm run test

NOTE:

Go to postman and run this endpoint: POST http://localhost:3000/api/v1/users/signin
using this sample resource:
```
{
  "email" : "dele@gmail.com",
  "password" : "dele1234"
}
```
Make sure to include your signed in token in the Bearer token
1. To add a resource: POST `http://localhost:3000/api/v1/resource`
using this sample resource :-
```
{
	"title": "Let it rain",
	"text": "hhhhhhhhhhhhhhhhhhhhhhhhs",
	"file": "httt",
	"service": "Jesus Celebration Service",
	"category": "Relevant information"
}
```
2. To update a resource: PATCH `http://localhost:3000/api/v1/resource`
using this sample resource :-
```
{
	"title": "Let it rain",
	"text": "hAPPY",
	"file": "httt",
	"service": "Jesus Celebration Service",
	"category": "Relevant information"
}
```
3. To get a resource: GET `http://localhost:3000/api/v1/resource/1`
4. To get all resources(everyone): GET `http://localhost:3000/api/v1/resource`
5. To get all resources for a specific user: GET `http://localhost:3000/api/v1/resources`
6. To delete a resource: DELETE `http://localhost:3000/api/v1/resource/1`
7. To get all songs: `http;//localhost:3000/api/v1/songs?category=Songs`

**What are the relevant pivotal tracker stories?**
<a href = "https://www.pivotaltracker.com/story/show/170073101"> #170071446</a>
